grails-app/services/org/pih/warehouse/data/PersonService.groovy

```diff
@@ -11,33 +11,136 @@ package org.pih.warehouse.data
+     */
+    private Person getPersonByEmail(String email) {
+        List<Person> people = Person.findAllByEmail(email)
+        if (people?.empty) {
+            return null
+        }
+
+        // We give preference to active people, so if there is one found, return it, otherwise return anyone.
+        return people.find{ it.active } ?: people.first()
     }
 
-    Person getPersonByNames(String combinedNames) {
+    /**
+     * @return The first person we can find who is active and has the given email
+     */
+    Person getActivePersonByEmail(String email) {
+        return Person.findByActiveAndEmail(true, email)
+    }
+
+    /**
+     * Fetch a user via their full name. We require the combined name to be provided (instead of first name
+     * and last name individually) because while internally we split the name so that it conforms to the firstName and
+     * lastName fields in the Person table, we want to eventually move away from forcing that name structure upon users.
+     *
+     * @return Returns the first person we can find who has the given email, giving preference to active people.
+     */
+    private Person getPersonByName(String combinedNames) {
         String[] names = extractNames(combinedNames)
-        return getPersonByNames(names)
+        List<Person> people = Person.findAllByFirstNameAndLastName(names[0], names[1])
+        if (people?.empty) {
+            return null
+        }
+
+        // We give preference to active people, so if there is one found, return it, otherwise return anyone.
+        return people.find{ it.active } ?: people.first()
+    }
+
+    /**
+     * Fetch an active user via their full name. We require the combined name to be provided (instead of first name
+     * and last name individually) because while internally we split the name so that it conforms to the firstName and
+     * lastName fields in the Person table, we want to eventually move away from forcing that name structure upon users.
+     *
+     * @param combinedNames The person's full name (first name + last name) combined into a single String.
+     * @return A Person who is active and has a firstName and lastName matching the combinedNames
+     */
+    Person getActivePersonByName(String combinedNames) {
+        if (StringUtils.isBlank(combinedNames)) {
+            return null
+        }
+
+        String[] names = extractNames(combinedNames)
+        return Person.findByActiveAndFirstNameAndLastName(true, names[0], names[1])
+    }
+
+    /**
+     * @param recipient Either a RFC822O internet/email address (ex: "Justin Miranda <justin@openboxes.com>")
+     *        or simply the recipient's name (ex: "Justin Miranda"). We accept both formats since either
+     *        format can be provided during data imports.
+     * @return A Person who has the provided name and email. Will be a new Person if they didn't previously exist.
+     */
+    Person getOrCreatePersonByRecipient(String recipient) {
```

Comment:
for the most part this method is copied from ShipmentService. It did refactor it though (and created tests for it)